src/mds/Server.cc

```diff
@@ -10710,3 +10758,309 @@ void Server::dump_reconnect_status(Formatter *f) const
   f->dump_stream("client_reconnect_gather") << client_reconnect_gather;
   f->close_section();
 }
+
+void Server::_readdir_diff(
```

Comment:
This is a large routine and should be split into smaller callable parts.

Reply:
Will do that when implementing explicit readdir_snapdiff API

Reply:
Did you miss updating this part in the latest push?